Source/Dafny/Verifier/Translator.ExpressionTranslator.cs

```diff
@@ -1882,7 +1882,9 @@ public ExpressionTranslator(ExpressionTranslator etran, string modifiesFrame)
             return disjunction;
           }
         }
-        return Boogie.Expr.Gt(Boogie.Expr.SelectTok(tok, TrExpr(s), elmtBox), Boogie.Expr.Literal(0));
+        var result = Boogie.Expr.Gt(Boogie.Expr.SelectTok(tok, TrExpr(s), elmtBox), Boogie.Expr.Literal(0));
+        result.tok = tok;
+        return result;
```

Comment:
This seems like it may be wrong for all calls into `Boogie.Expr....`. A possibility would be to fix all of them by changing line 172 of `$Boogie/Source/Core/AbsyExpr.cs` to replace `Token.NoToken` with `e0.tok`.